x-pack/plugins/apm/public/components/app/mobile/service_overview/metrics_map/embedded_map.tsx

```diff
@@ -125,6 +125,30 @@ function EmbeddedMapComponent({
     }
   }, [embeddable, embeddableRoot]);
 
+  useEffect(() => {
+    const setLayerList = async () => {
+      if (embeddable) {
+        let layerList;
+        switch (selectedMap) {
+          case MapTypes.HTTP:
+            layerList = await getHttpRequestsLayerList(maps);
+
+            break;
+          case MapTypes.SESSIONS:
+            layerList = await getSessionMapLayerList(maps);
+            break;
+          default:
+            layerList = await getHttpRequestsLayerList(maps);
+        }
+        // @ts-expect-error
+        await embeddable.setLayerList(layerList);
+        await embeddable.reload();
```

Comment:
Can't we run it in parallel? Maybe even add the `layerList` if you extract the switch to another function and return a promise.

```suggestion
        await Promise.all([getLayerList(maps), embeddable.setLayerList(layerList), embeddable.reload()];
```

Reply:
I need the `LayerList` before for calling `seLayerList`

```
const layerList = await getLayerList({ selectedMap, maps });
 await Promise.all([
    embeddable.setLayerList(layerList),
    embeddable.reload(),
]);
```
is there any better way?